### PR TITLE
permissions: migrate to new auth system and accept credentials

### PR DESCRIPTION
**Pull request by Rugvip**
On top of #23054, work towards supporting the new auth services

### CODE REVIEW

**Review by freben — APPROVED** [review]
super nice, preapproved but some small notes

File: plugins/permission-node/src/ServerPermissionClient.ts [hunk 1]
```diff
@@ -62,58 +68,88 @@ export class ServerPermissionClient implements PermissionEvaluator {
...
   }
 
   async authorizeConditional(
     queries: QueryPermissionRequest[],
-    options?: EvaluatorRequestOptions,
+    options?: PermissionsServiceRequestOptions,
   ): Promise<PolicyDecision[]> {
-    return (await this.isEnabled(options?.token))
-      ? this.permissionClient.authorizeConditional(queries, options)
+    return (await this.shouldPermissionsBeApplied(options))
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/permission-node/src/ServerPermissionClient.ts]: Yep pushed something to that effect, didn't quite feel the bonus bool was worth it tho

**Comment by Rugvip** [comment]
Added a couple of tests for the `credentials` cases now too

File: plugins/permission-node/src/ServerPermissionClient.ts [hunk 2]
```diff
@@ -62,58 +68,92 @@ export class ServerPermissionClient implements PermissionEvaluator {
...
+      return false;
+    }
+
+    let credentials: BackstageCredentials;
+    if (options && 'credentials' in options) {
+      credentials = options.credentials;
+    } else {
+      if (!options?.token) {
+        return true;
+      }
```
Comment (Joonpark13) [inline plugins/permission-node/src/ServerPermissionClient.ts:146]: Minor nit: structuring the conditionals this way, we lose the semantic `isValidServerToken` var name, which I think was very helpful in terms of readability (which is arguably more important now that we have multiple levels of nesting)

**Review by Joonpark13 — APPROVED** [review]

File: plugins/permission-node/src/ServerPermissionClient.ts (hunk 2 above)
Comment (Rugvip) [reply plugins/permission-node/src/ServerPermissionClient.ts:146]: hmm, not sure I follow how you'd want to structure it tbh? the server token check is now down in the `isPrincipal` call, which feels straight-forward enough at this point

**Pull request merged**